synapse/lib/types.py

```diff
@@ -694,26 +702,34 @@ def _preNormHex(self, text):
     def _storLiftEq(self, cmpr, valu):
 
         if isinstance(valu, str):
-            valu = self._preNormHex(valu)
+            valu = self._preNormHexStr(valu)
             if valu.endswith('*'):
                 return (
                     ('^=', valu[:-1], self.stortype),
                 )
 
+        if isinstance(valu, int):
+            valu = self._preNormHexInt(valu)
+            if len(valu) % 2 != 0:
+                valu = f'0{valu}'
+
         return self._storLiftNorm(cmpr, valu)
 
     def _storLiftPref(self, cmpr, valu):
-        valu = self._preNormHex(valu)
+        if isinstance(valu, int):
```

Comment:
Feels like this ( and the above ) conditional logic could be replaced with just a `self.norm()` call which does the same thing, no?

Reply:
No, the `_preNormHex*` functions just make sure the valu is in the correct format. If we did a full `norm()` on it, it would get checked against `self._size` and potentially get zeropadded. We don't want this things because this is just a prefix check.